### PR TITLE
add rhessys files

### PR DESCRIPTION
**Pull request by DavidChoi76**
added rhessys.py 
update rhessysSupervisor.py
added Dockerfile and Definition file

### CODE REVIEW

File: cybergis/rhessys.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+
+class RHESSysKeelingSBatchScript(KeelingSBatchScript):
+    file_name = "rhessys.sbatch"
+    SCRIPT_TEMPLATE = RHESSys_SBATCH_SCRIPT_TEMPLATE
+
+    def __init__(self, walltime, ntasks,
+                 *args, **kargs):
+        super().__init__(walltime, ntasks, *args, **kargs)
+        self.remote_singularity_img_path = "/home/yc5ef/rhessys_20210201.sif"
```
Comment (zhiyuli) [inline cybergis/rhessys.py]: @DavidChoi76 we need to change to use path on keeling: 
"/data/keeling/a/cigi-gisolve/simages/rhessys_20210201.sif"

Also, what is the base OS version in your singularity image? ubuntu 18 or 16?

File: cybergis/rhessys.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+
+class RHESSysCometSBatchScript(CometSBatchScript):
+    file_name = "rhessys.sbatch"
+    SCRIPT_TEMPLATE = RHESSys_SBATCH_SCRIPT_TEMPLATE
+
+    def __init__(self, walltime, ntasks, *args, **kargs):
+        super().__init__(walltime, ntasks, *args, **kargs)
+
+        self.remote_singularity_img_path = "/home/yc5ef/rhessys_20210201.sif"
```
Comment (zhiyuli) [inline cybergis/rhessys.py]: @DavidChoi76  same here. change to comet path: 
"/home/cybergis/SUMMA_IMAGE/rhessys_20210201.sif"

File: cybergis/rhessys.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+class RHESSysUserScript(BaseScript):
+    SCRIPT_TEMPLATE = RHESSys_USER_SCRIPT_TEMPLATE
+    file_name = "runRHESSys.py"
+
+
+class RHESSysKeelingJob(KeelingJob):
+    job_name = "RHESSys"
+    sbatch_script_class = RHESSysKeelingSBatchScript
+
+    def __init__(
```
Comment (zhiyuli) [inline cybergis/rhessys.py]: @DavidChoi76 if this function does not override anything in baseclass, we should be able to remove it.

File: singularity_def/rhessys/Dockerfile_rhessysEC7.2 [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    apt-get install -y --no-install-recommends \
+    software-properties-common python-software-properties \
+    ca-certificates \
+    git \
+    make \
+    liblapack-dev \
+    vim
+
+# set environment variables for docker build
+ENV F_MASTER /code
```
Comment (zhiyuli) [inline singularity_def/rhessys/Dockerfile_rhessysEC7.2]: @DavidChoi76 let's clean this up a little bit. Better to remove all SUMMA staff

File: singularity_def/rhessys/Dockerfile_rhessysEC7.2 [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ENV FC gcc
+ENV FC_EXE gcc
+ENV INCLUDES -I/usr/include
+ENV LIBRARIES '-L/usr/lib -lnetcdff -llapack -lblas'
+
+RUN git clone https://github.com/NCAR/summa.git /tmp/summa \
```
Comment (zhiyuli) [inline singularity_def/rhessys/Dockerfile_rhessysEC7.2]: summa leftover

File: singularity_def/rhessys/Dockerfile_rhessysEC7.2 [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    && bash Miniconda3-py37_4.8.3-Linux-x86_64.sh -b -p /opt/miniconda3 \
+    && rm Miniconda3-py37_4.8.3-Linux-x86_64.sh \
+    && chown -R docker:docker /opt/miniconda3
+    
+#Set environment variables
+ENV PATH="/opt/miniconda3/bin:${PATH}"
+
+# Install SUMMA conda env
+
+RUN /opt/miniconda3/bin/python -m pip install --quiet git+https://github.com/DavidChoi76/pyrhessys.git
```
Comment (zhiyuli) [inline singularity_def/rhessys/Dockerfile_rhessysEC7.2:44]: @DavidChoi76  is this the only lib that needs to be installed for rhessys? it has both PyRhessys and Rhessys binary?

File: singularity_def/rhessys/Dockerfile_rhessysEC7.2 [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+RUN /opt/miniconda3/bin/conda install -c conda-forge mpi4py=3.0.3=py37h0c5ec45_2
+
+RUN mkdir -p /workspace && chmod -R 777 /workspace
+RUN chmod -R 777 /home/docker
+RUN chmod -R 777 /opt/miniconda3
+
+RUN mkdir -p /home/jovyan/work/workspace
+RUN chmod -R 777 /home
+
+# create a symbolic link to summa.exe being consistent with old summa singularity images
```
Comment (zhiyuli) [inline singularity_def/rhessys/Dockerfile_rhessysEC7.2]: summa leftover

**Review by zhiyuli — CHANGES_REQUESTED** [review]
@DavidChoi76 I finished first review of this repo. thanks

File: cybergis/rhessys.py (hunk 1 above)
Comment (DavidChoi76) [reply cybergis/rhessys.py]: I used `ubuntu:xenial` like SUMMA Dockerfile. You can check Dockerfile and Singularity file in https://github.com/cybergis/Jupyter-xsede/tree/rhessys/singularity_def/rhessys

**Review by zhiyuli — APPROVED** [review]

**Pull request merged**